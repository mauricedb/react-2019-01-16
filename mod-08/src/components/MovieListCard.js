import React from 'react';

export default function MovieListCard({ movie, onMovieClicked }) {
  const result = (
    <div className="movie" onClick={() => onMovieClicked(movie)}>
      <h4>{movie.title}</h4>
      <img src={movie.image} alt={movie.title} />
      <div>{movie.overview}</div>
    </div>
  );

  return result;
}

MovieListCard.displayName = 'MovieListCard';
