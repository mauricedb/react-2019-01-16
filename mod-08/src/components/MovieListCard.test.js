import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import MovieListCard from './MovieListCard';

test('The MovieListCard can render', () => {
  const onMovieClicked = jest.fn();

  const wrapper = render(
    <MovieListCard
      movie={{
        title: 'The movie'
      }}
      onMovieClicked={onMovieClicked}
    />
  );

  fireEvent.click(wrapper.getByText('The: The movie'));

  expect(onMovieClicked).toHaveBeenCalledTimes(1);
});

test('snapshot', () => {
  expect({ x: 1, y: 2, z: 3 }).toMatchSnapshot();
});
