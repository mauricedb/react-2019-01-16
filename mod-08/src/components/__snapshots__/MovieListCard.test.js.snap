// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "x": 1,
  "y": 2,
  "z": 3,
}
`;
